apps/api/src/app/auth/services/passport/apikey.strategy.ts

```diff
@@ -0,0 +1,24 @@
+import { HeaderAPIKeyStrategy } from 'passport-headerapikey';
+import { PassportStrategy } from '@nestjs/passport';
+import { Injectable, Logger } from '@nestjs/common';
+import { AuthService } from '@novu/application-generic';
+import { IJwtPayload } from '@novu/shared';
+
+@Injectable()
+export class ApiKeyStrategy extends PassportStrategy(HeaderAPIKeyStrategy) {
+  constructor(private readonly authService: AuthService) {
+    super(
+      { header: 'Authorization', prefix: 'ApiKey ' },
+      true,
+      (apikey: string, verified: (err: Error | null, user?: IJwtPayload | false) => void) => {
+        this.authService.validateApiKey(apikey).then((user) => {
```

Comment:
Should we add a catch here?

Reply:
Good question! I'll add investigate whether an unhandled error here results in an `Unauthenticated` error and address accordingly.

Reply:
I simulated an error on the `validateApiKey` method, unhandled errors are not managed internally by passport. I've added a `.catch`. As a side, my preference would have been to use async/await but the `passport-headerapikey` lib does not support this.

_Before with unhandled error_
<img width="1370" alt="image" src="https://github.com/novuhq/novu/assets/32132657/c8adb022-9426-4b6b-bc51-74c67ac277a5">

_After with `.catch` handler, for simulated 500 error_
<img width="1366" alt="image" src="https://github.com/novuhq/novu/assets/32132657/cc521f93-5145-463c-868b-ce447c57b526">

_After with `.catch` handle, for 400 auth error_

Reply:
i wanted to suggest an async-await pattern as well because i pretty much hate nested promises, but then i saw that they do not expect an async function :(